### ISSUE TITLE
Offer Chat + Core Beacon as Separate Artifacts

### ISSUE THREAD

**Issue by bvirlet** [helpscout/beacon-ios-sdk#60]
* Beacon install method: [x] Carthage, [ ] CocoaPods, [ ] Manual

I'm concerned about the future Beacon SDK 2.0 increase of dependencies. We strive to have as little dependencies as possible (especially binary ones that we can't fix or know exactly what's in it) and I see that Beacon SDK 2.0 is adding 4 dependencies: https://github.com/helpscout/beacon-ios-sdk/blame/dc28b3c345f96d7d93eaa2d3bf29abbbd3976dc3/Cartfile.resolved.

We are not interested by the chat feature at all, so we would appreciate a way to integrate the Beacon SDK without all these extra dependencies.

**Comment by shaunmcilroy** [comment]
Hey @bvirlet ,

Thanks for the feedback, appreciate you taking the time to post your thoughts for the team and the wider development community. I can appreciate that you won't wish to make use of the Beacon Chat feature when it's fully released however we have no plans to offer multiple versions of the SDK where those dependencies which enable that feature won't be present at this time.

In order to create a forked version of our SDK where the newer dependencies required for Beacon Chat are not included it would require our development team to develop and test for two versions of the SDK with every future release. This would limit the speed by which we can add more improvements and features down the line - such as our upcoming visual refresh for iOS which brings Beacon [in-line with the web version](https://c.hlp.sc/p9u76be6) and the Android version [which is in progress](https://github.com/helpscout/beacon-android-sdk-sample/blob/master/CHANGELOG.md). With that in mind we have no plans at this time to take such an approach in the future.

We'll definitely pass your thoughts onto the product leadership for future consideration and keep you informed with any updates. You can also keep tabs for any changes via our changelog to allow you to determine whether Beacon is the right choice for you and your customers, and I hope that you keep any feedback coming as you continue to offer Beacon as a support option for your audience.

Thanks again,
Shaun

**Comment by bvirlet** [comment]
Thanks for your reply, @shaunmcilroy. Without forking, there are ways to split a SDK in multiple frameworks, such as CocoaPod's subspecs. 

Or just having multiple frameworks so that we don't have to include the entire functionality. Eg. you can have BeaconSDKCore, BeaconSDKChat. If you include both, you get the entire feature set. But you can choose to only include BeaconSDKCore.

**Comment by ScottPetit** [comment]
Hey @bvirlet I wanted to reach out and let you know where we ended up with the latest version (2.0.0) of the SDK.  I totally get not wanting to introduce additional dependencies, and due to this feedback and some other considerations, we've made all of Beacon's dependencies internal.  So now going from `v1.0.5` to `v2.0.0` reduces the frameworks required to be built for Beacon from 2 to 1.

I know this isn't exactly what you were requesting with this issue, but hopefully addresses one of the main problems with the previous way we handled dependencies.
